File: avatarAndDat/models/nft_market_table.go
package models

import "time"

type NftMarketTable struct {
	NftLdefIndex string `orm:"pk;unique"`
	SellerWalletId string
	SellerNickname string
	MpId string
	NftAdminId string
	Price int
	Qty int
	NumSold int
	Active bool
	ActiveTicker string
	AllowAirdrop bool
	CreatorPercent int
	LyricsWriterPercent int
	SongComposerPercent int
	PublisherPercent int
	UserPercent int
	Timestamp time.Time `orm:"auto_now_add;type(datetime)"`
}

func (this *NftMarketTable) TableIndex() [][]string {
	return [][]string {
		[]string{"NftAdminId"},
		[]string{"SellerWalletId"},
		[]string{"SellerNickname"},
	}
}

File: py_backend/model/nft_market_table.py
# coding: utf8

from .core import db


class NftMarketTable(db.Model):
    NftLdefIndex = db.Column(db.String, primary_key=True)
    MpId = db.Column(db.String)
    NftAdminId = db.Column(db.String)
    Price = db.Column(db.Integer)
    Qty = db.Column(db.Integer)
    NumSold = db.Column(db.Integer)
    Active = db.Column(db.Boolean)
    ActiveTicker = db.Column(db.String)

    def __repr__(self):
        return '<NftLdefIndex %r>' % self.NftLdefIndex

    def __init__(self,
                 NftLdefIndex: str,
                 MpId: str,
                 NftAdminId: str,
                 Price: int,
                 Qty: int,
                 NumSold: int,
                 Active: bool,
                 ActiveTicker: str):
        self.NftLdefIndex = NftLdefIndex
        self.MpId = MpId
        self.NftAdminId = NftAdminId
        self.Price = Price
        self.Qty = Qty
        self.NumSold = NumSold
        self.Active = Active
        self.ActiveTicker = ActiveTicker
